pkg/adminconfig/rego_file_evaluator_test.go

```diff
@@ -74,13 +78,134 @@ func NewEvaluator() *adminconfig.RegoPolicyEvaluator {
+		}
+
+		# High Priority Production Workloads - read
+		config[{"read": decision}] {
+			input.request.usage.read == true
+			input.workload.properties.stage == "PROD"
+			input.workload.properties.priority == "high"
+			dataset_region := input.request.dataset.geography
+			workload_region := input.workload.cluster.metadata.region	
+			dataset_region != workload_region		
+			policy := {"description": "focus on high performance"}
+		    clusters := { "metadata.region" : [ workload_region ] }
+			decision := {"policy": policy, "deploy": true, "restrictions": {"clusters": clusters}}
+		}
+
+		# High Priority Production Workloads - copy
+		config[{"copy": decision}] {
+			input.request.usage.read == true
+			input.workload.properties.stage == "PROD"
+			input.workload.properties.priority == "high"
+			dataset_region := input.request.dataset.geography
+			workload_region := input.workload.cluster.metadata.region	
+			dataset_region != workload_region		
+			policy := {"description": "focus on high performance"}
+		    accounts := [data.infrastructure.storageaccounts.values[i].id | data.infrastructure.storageaccounts.values[i].region == workload_region; 
+																	 		data.infrastructure.storageaccounts.values[i].type == "object-storage" ]
+			decision := {"policy": policy, "deploy": true, "restrictions": {"storageaccounts": {"values.id": accounts}}}
+		}
+
+		# Transform
+		config[{"transform": decision}] {
+			policy := {"ID": "transform-geo", "description":"Governance based transformations must take place in the geography where the data is stored"}
+			clusters := { "metadata.region" : [ input.request.dataset.geography ] }
+			decision := {"policy": policy, "restrictions": {"clusters": clusters}}
+		}
+
+	`
+	// Compile the module. The keys are used as identifiers in error messages.
+	compiler, err := ast.CompileModules(map[string]string{
+		"example.rego": module,
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	data := `{
+		"infrastructure": {
+			"bandwidth": {
+				"units": "GB/sec",
+				"scale": {},
+				"values": {
+					"Netherlands": {"Netherlands": "L", "Romania": "M", "Australia": "S"},
+					"Romania": {"Romania": "L", "Netherlands": "M", "Australia": "S"},
+					"Australia": {"Australia": "L", "Romania": "S", "Netherlands": "S"}
+				}
+			},
+			"storageaccounts": {
+				"units": "dollar",
+				"scale": {},
+				"values": [
+					{"id": "Netherlands-storage", "region": "Netherlands", "type": "relational-database", "cost": "100"},
+					{"id": "Netherlands-storage", "region": "Netherlands", "type": "object-storage", "cost": "100"},
+					{"id": "Romania-storage", "region": "Romania", "type": "object-storage", "cost": "80"},
+					{"id": "Romania-storage", "region": "Romania", "type": "relational-database", "cost": "80"},
+					{"id": "Australia-storage", "region": "Australia", "type": "relational-database", "cost": "20"},
+					{"id": "Australia-storage", "region": "Australia", "type": "object-storage", "cost": "90"}
```

Comment:
We need unique id here. For example, Australia-object-store and Australia-DB-storage

Reply:
Changed